src/core/songloader.cpp

```diff
@@ -156,10 +156,8 @@ SongLoader::Result SongLoader::LoadLocalPartial(const QString& filename) {
 SongLoader::Result SongLoader::LoadAudioCD() {
 #ifdef HAVE_AUDIOCD
   CddaSongLoader* cdda_song_loader = new CddaSongLoader;
-  connect(cdda_song_loader, SIGNAL(SongsDurationLoaded(SongList)), this,
-          SLOT(AudioCDTracksLoadedSlot(SongList)));
-  connect(cdda_song_loader, SIGNAL(SongsMetadataLoaded(SongList)), this,
-          SLOT(AudioCDTracksTagsLoaded(SongList)));
+  connect(cdda_song_loader, SIGNAL(SongsUpdated(SongList, bool)), this,
+          SLOT(AudioCDTracksLoadedSlot(SongList, bool)));
```

Comment:
While I think that unifying these is mostly correct, I wonder if we should have a separate Finished signal where the connected slot would just delete the object and handle any errors. I think it might be a little clearer than the boolean passed to SongsUpdated.

Reply:
That's a good idea and I have changed it accordingly.